test/CSSTransition-test.js

```diff
@@ -355,19 +375,20 @@ describe('CSSTransition', () => {
               <CSSTransition
                 key={text}
```

Comment:
When you change the key of a child in `TransitionGroup`, _another_ copy of the element is created and starts entering, while the old one starts exiting before being removed form the DOM. I created a CodeSandbox based on this test so you can see:

[![Edit objective-jackson-30ghb](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/objective-jackson-30ghb?fontsize=14&hidenavigation=1&theme=dark)

One way to fix this test is by managing two refs, one for each key.

Btw, this might trip some people up when migrating to `nodeRef`, so it's important that we mention that.